diff --git a/drivers/clk/sunxi-ng/ccu-sun50i-a64.c b/drivers/clk/sunxi-ng/ccu-sun50i-a64.c
index b623c8150b4f..1f47e4954e9d 100644
--- a/drivers/clk/sunxi-ng/ccu-sun50i-a64.c
+++ b/drivers/clk/sunxi-ng/ccu-sun50i-a64.c
@@ -183,7 +183,7 @@ static struct ccu_nkm pll_mipi_clk = {
        .n              = _SUNXI_CCU_MULT(8, 4),
        .k              = _SUNXI_CCU_MULT_MIN(4, 2, 2),
        .m              = _SUNXI_CCU_DIV(0, 4),
-       .min_rate       = 300000000,            /* Actual rate is 500MHz */
+       .min_rate       = 330000000,            /* Actual rate is 500MHz */
        .common         = {
                .reg            = 0x040,
                .hw.init        = CLK_HW_INIT("pll-mipi", "pll-video0",
diff --git a/drivers/clk/sunxi-ng/ccu_nkm.c b/drivers/clk/sunxi-ng/ccu_nkm.c
index 096ff4f4839a..6c7a705b8bfb 100644
--- a/drivers/clk/sunxi-ng/ccu_nkm.c
+++ b/drivers/clk/sunxi-ng/ccu_nkm.c
@@ -125,7 +125,7 @@ static unsigned long ccu_nkm_round_rate(struct ccu_mux_internal *mux,
        if (nkm->common.features & CCU_FEATURE_FIXED_POSTDIV)
                rate *= nkm->fixed_post_div;
 
-       if (rate < nkm->min_rate) {
+       if (rate <= nkm->min_rate) {
                rate = nkm->min_rate;
                return rate;
        }