From 4e7a61eb3439ec039b171edf238d52cb7879a83f Mon Sep 17 00:00:00 2001
From: Luca Weiss <luca@z3ntu.xyz>
Date: Fri, 12 Apr 2019 22:05:28 +0200
Subject: [PATCH] Remove unused libnm-glib dependency

---
 CMakeLists.txt | 1 -
 debian/control | 1 -
 2 files changed, 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index fddf076c..a7f7f297 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -45,7 +45,6 @@ pkg_search_module(GOBJECT REQUIRED gobject-2.0)
 pkg_search_module(GLIB REQUIRED glib-2.0)
 pkg_check_modules(GIO REQUIRED gio-2.0 gio-unix-2.0)
 pkg_search_module(UPOWER_GLIB REQUIRED upower-glib)
-pkg_search_module(LIBNM_GLIB REQUIRED libnm-glib)
 pkg_search_module(ACCOUNTSSERVICE REQUIRED accountsservice)
 pkg_search_module(GEONAMES REQUIRED geonames)
 pkg_search_module(ICU REQUIRED icu-i18n)
diff --git a/debian/control b/debian/control
index a8cea9a3..fd0a68a1 100644
--- a/debian/control
+++ b/debian/control
@@ -17,7 +17,6 @@ Build-Depends: dbus-test-runner,
                libglib2.0-dev (>= 2.37.92),
                libevdev-dev,
                libicu-dev,
-               libnm-glib-dev,
                libpolkit-agent-1-dev,
                libqmenumodel-dev,
                libtrust-store-dev,
-- 
2.21.0

