diff --git a/external/process-cpp-minimal/include/core/posix/standard_stream.h b/external/process-cpp-minimal/include/core/posix/standard_stream.h
index f50a9b7..4a67a80 100644
--- a/external/process-cpp-minimal/include/core/posix/standard_stream.h
+++ b/external/process-cpp-minimal/include/core/posix/standard_stream.h
@@ -23,6 +23,10 @@

 #include <cstdint>

+#undef stdin
+#undef stdout
+#undef stderr
+
 namespace core
 {
 namespace posix
diff --git a/external/process-cpp-minimal/src/core/posix/signal.cpp b/external/process-cpp-minimal/src/core/posix/signal.cpp
index fd90047..e66f3e6 100644
--- a/external/process-cpp-minimal/src/core/posix/signal.cpp
+++ b/external/process-cpp-minimal/src/core/posix/signal.cpp
@@ -155,7 +155,7 @@ public:
             {
                 auto result = ::read(scope.signal_fd, signal_info, sizeof(signal_info));

-                for (uint i = 0; i < result / sizeof(signalfd_siginfo); i++)
+                for (unsigned int i = 0; i < result / sizeof(signalfd_siginfo); i++)
                 {
                     if (has(static_cast<core::posix::Signal>(signal_info[i].ssi_signo)))
                     {
diff --git a/src/anbox/logger.cpp b/src/anbox/logger.cpp
index 93b7e10..88a5e09 100644
--- a/src/anbox/logger.cpp
+++ b/src/anbox/logger.cpp
@@ -190,7 +190,7 @@ std::ostream& operator<<(std::ostream& strm, anbox::Logger::Severity severity) {
     case anbox::Logger::Severity::kFatal:
       return strm << "FF";
     default:
-      return strm << static_cast<uint>(severity);
+      return strm << static_cast<unsigned int>(severity);
   }
 }

diff --git a/external/process-cpp-minimal/src/CMakeLists.txt b/external/process-cpp-minimal/src/CMakeLists.txt
index dec4659..20c9053 100644
--- a/external/process-cpp-minimal/src/CMakeLists.txt
+++ b/external/process-cpp-minimal/src/CMakeLists.txt
@@ -46,4 +46,5 @@ target_link_libraries(
 
   ${Boost_LIBRARIES}
   ${CMAKE_THREAD_LIBS_INIT}
+  execinfo
 )

--
From e6a06d1bf3af27400aff49772f1b04ebdb944168 Mon Sep 17 00:00:00 2001
From: Grant Miller <GrantM11235@gmail.com>
Date: Thu, 24 Jan 2019 10:07:07 -0600
Subject: [PATCH 2/3] Change __dev_t to dev_t

---
 src/anbox/container/lxc_container.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/anbox/container/lxc_container.cpp b/src/anbox/container/lxc_container.cpp
index 5790fe3..0dbe181 100644
--- a/src/anbox/container/lxc_container.cpp
+++ b/src/anbox/container/lxc_container.cpp
@@ -77,11 +77,11 @@ constexpr const char *lxc_config_log_file_key{"lxc.log.file"};
 constexpr const char *lxc_config_apparmor_profile_key{"lxc.apparmor.profile"};
 #endif
 
-constexpr int device_major(__dev_t dev) {
+constexpr int device_major(dev_t dev) {
   return int(((dev >> 8) & 0xfff) | ((dev >> 32) & (0xfffff000)));
 }
 
-constexpr int device_minor(__dev_t dev) {
+constexpr int device_minor(dev_t dev) {
   return int((dev & 0xff) | ((dev >> 12) & (0xffffff00)));
 }
 } // namespace

