diff --git a/examples/src/CMakeLists.txt b/examples/src/CMakeLists.txt
index 7ad6aa8..203b698 100644
--- a/examples/src/CMakeLists.txt
+++ b/examples/src/CMakeLists.txt
@@ -8,7 +8,10 @@ target_link_libraries(example_power
 install(TARGETS example_power DESTINATION ${DST_LIB}/statefs)
 
 add_library(example_statefspp SHARED example_statefspp.cpp)
-target_link_libraries(example_statefspp statefs-pp)
+target_link_libraries(example_statefspp
+    statefs-pp
+    ${CMAKE_THREAD_LIBS_INIT}
+)
 install(TARGETS example_statefspp DESTINATION ${DST_LIB}/statefs)
 
 set(POWER_SRC basic.c)
