diff --git a/drivers/ril/ril.c b/drivers/ril/ril.c
index eaaec44..8c7d3ab 100755
--- a/drivers/ril/ril.c
+++ b/drivers/ril/ril.c
@@ -9,7 +9,7 @@
 #include <linux/uaccess.h>
 #include <linux/workqueue.h>
 #include <linux/wakelock.h>
-#include <../../arch/arm/mach-tegra/include/mach/board-cardhu-misc.h>
+#include "../../arch/arm/mach-tegra/include/mach/board-cardhu-misc.h"
 
 #include "pm-irq.h"
 #include "ril.h"
